src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/ProxyStreamFactoryBuilder.java

```diff
@@ -155,51 +161,79 @@ public StreamFactoryBuilder setAccumulatorSupplier(
         return this;
     }
 
+    @Override
+    public StreamFactoryBuilder setExecutor(
+        SignalingExecutor executor)
+    {
+        this.executor = executor;
+        return this;
+    }
+
     @Override
     public StreamFactory build()
     {
         final HttpCacheCounters counters = new HttpCacheCounters(supplyCounter, supplyAccumulator);
 
-        if (cache == null)
+        if (emulatedBudgetManager == null && defaultBudgetManager == null)
         {
-            budgetManager = new BudgetManager();
+            emulatedBudgetManager = new BudgetManager();
+            defaultBudgetManager =  new BudgetManager();
             final int httpCacheCapacity = config.cacheCapacity();
             final int httpCacheSlotCapacity = config.cacheSlotCapacity();
             this.cacheBufferPool = new Slab(httpCacheCapacity, httpCacheSlotCapacity);
             this.requestBufferPool = new HeapBufferPool(config.maximumRequests(), httpCacheSlotCapacity);
+        }
+
+        if (cacheEntries == null)
+        {
+            cacheEntries = supplyAccumulator.apply("http-cache.cache.entries");
+        }
```

Comment:
Why guard with `null` check?

Reply:
We don't want to recreate it right?

Reply:
`supplyAccumulator` can be called multiple times, returning same instances - also, `build()` is only called once.